src/libraries/System.Private.CoreLib/src/System/Enum.cs

```diff
@@ -345,43 +345,83 @@ public static bool IsDefined(Type enumType, object value)
         public static object Parse(Type enumType, string value) =>
             Parse(enumType, value, ignoreCase: false);
 
+        public static object Parse(Type enumType, ReadOnlySpan<char> value) =>
+            Parse(enumType, value, ignoreCase: false);
+
         public static object Parse(Type enumType, string value, bool ignoreCase)
         {
             bool success = TryParse(enumType, value, ignoreCase, throwOnFailure: true, out object? result);
             Debug.Assert(success);
             return result!;
         }
 
+        public static object Parse(Type enumType, ReadOnlySpan<char> value, bool ignoreCase)
+        {
+            bool success = TryParse(enumType, value, ignoreCase, throwOnFailure: true, out object? result);
+            Debug.Assert(success);
+            return result!;
+        }
+
         public static TEnum Parse<TEnum>(string value) where TEnum : struct =>
             Parse<TEnum>(value, ignoreCase: false);
 
+        public static TEnum Parse<TEnum>(ReadOnlySpan<char> value) where TEnum : struct =>
+           Parse<TEnum>(value, ignoreCase: false);
+
         public static TEnum Parse<TEnum>(string value, bool ignoreCase) where TEnum : struct
         {
             bool success = TryParse<TEnum>(value, ignoreCase, throwOnFailure: true, out TEnum result);
             Debug.Assert(success);
             return result;
         }
 
+        public static TEnum Parse<TEnum>(ReadOnlySpan<char> value, bool ignoreCase) where TEnum : struct
+        {
+            bool success = TryParse<TEnum>(value, ignoreCase, throwOnFailure: true, out TEnum result);
+            Debug.Assert(success);
+            return result;
+        }
+
         public static bool TryParse(Type enumType, string? value, out object? result) =>
             TryParse(enumType, value, ignoreCase: false, out result);
 
+        public static bool TryParse(Type enumType, ReadOnlySpan<char> value, out object? result) =>
+          TryParse(enumType, value, ignoreCase: false, out result);
+
         public static bool TryParse(Type enumType, string? value, bool ignoreCase, out object? result) =>
             TryParse(enumType, value, ignoreCase, throwOnFailure: false, out result);
 
+        public static bool TryParse(Type enumType, ReadOnlySpan<char> value, bool ignoreCase, out object? result) =>
+            TryParse(enumType, value, ignoreCase, throwOnFailure: false, out result);
+
         private static bool TryParse(Type enumType, string? value, bool ignoreCase, bool throwOnFailure, out object? result)
+        {
+            if (value == null)
+            {
+                if (throwOnFailure)
+                {
+                    throw new ArgumentNullException(nameof(value));
+                }
+                result = null;
+                return false;
+            }
+
+            ReadOnlySpan<char> valueSpan = value.AsSpan();
+
+            return TryParse(enumType, valueSpan, ignoreCase, throwOnFailure, out result);
```

Comment:
```suggestion
            return TryParse(enumType, value.AsSpan(), ignoreCase, throwOnFailure, out result);
```
NIT: you could have the AsSpan call happen as part of the method invocation